test/ffmpeg-vaapi/transcode/transcoder.py

```diff
@@ -0,0 +1,68 @@
+###
+### Copyright (C) 2018-2019 Intel Corporation
+###
+### SPDX-License-Identifier: BSD-3-Clause
+###
+
+from ....lib import *
+from ..util import *
+import os
+
+@slash.requires(have_ffmpeg)
+@slash.requires(have_ffmpeg_vaapi_accel)
+class TranscoderTest(slash.Test):
+  def before(self):
+    self.refctx = []
+
+  def transcode_1to1(self):
+    self.decoded = get_media()._test_artifact(
+      "{case}_{width}x{height}_{mode}.{dstextension}".format(**vars(self)))
+    if vars(self).get("mode", None) == 'hwhw':
+      self.output = call(
+        "ffmpeg -hwaccel vaapi -hwaccel_device /dev/dri/renderD128 -v verbose"
+        "  -hwaccel_output_format vaapi -i {source} -an -c:v {mcodec}"
+        " -vframes {frames} -y {decoded}".format(**vars(self)))
+    elif  vars(self).get("mode", None) == 'hwsw':
+      self.output = call(
+        "ffmpeg -hwaccel vaapi -hwaccel_device /dev/dri/renderD128 -v verbose"
+        "  -hwaccel_output_format vaapi -i {source}  -vf 'hwdownload,format=nv12' -an -c:v {mcodec}"
+        " -vframes {frames} -y {decoded}".format(**vars(self)))
+    elif  vars(self).get("mode", None) == 'swhw':
+      self.output = call(
+        "ffmpeg -vaapi_device /dev/dri/renderD128 -v verbose "
+        "-i {source}  -vf 'hwupload,format=nv12' -an -c:v {mcodec}"
```

Comment:
Oops, I missed this originally.  The -vf syntax here is backwards.  It should be `-vf 'format=nv12,hwupload'`  Please submit a new pull-request to fix this (since this PR is already merged).

Reply:
thank you and https://github.com/intel/vaapi-fits/pull/30 created for fixing this.